addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseThingHandler.java

```diff
@@ -0,0 +1,350 @@
+		updateStatus(ThingStatus.OFFLINE);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void dispose() {
+		logger.debug("Thing {} disposed.", getThing().getUID());
+
+		this.setThingRefreshed(false);
+		super.dispose();
+	}
+
+	/**
+	 * Method to Refresh Thing Handler.
+	 */
+	public void refreshThing() {
+
+		if(getDSCAlarmBridgeHandler() != null) {
+			updateStatus(dscAlarmBridgeHandler.getThing().getStatus());
+			logger.debug("refreshThing(): Bridge '{}' Found for Thing '{}'!",dscAlarmBridgeHandler.getThing().getUID(), this.getThing().getUID());
+
+			Thing thing = getThing();
+			List<Channel> channels = thing.getChannels();
+			logger.debug("refreshThing(): Refreshing Thing - {}",thing.getUID());
+			
+			this.properties = new DSCAlarmProperties();
+			
+			for(Channel channel : channels) {
+				updateChannel(channel.getUID());
+				//logger.debug("refreshThing(): Channel Initialized - {}",channel.getUID());
+			}
+	
+			if(dscAlarmThingType.equals(DSCAlarmThingType.PANEL)) {
+					dscAlarmBridgeHandler.setUserCode(getUserCode());
+			}
+	
+			this.setThingRefreshed(true);
+		}		
+		logger.debug("refreshThing(): Thing Refreshed - {}",thing.getUID());
+	}
+
+	/**
+	 * Get the Bridge Handler for the DSC Alarm.
+	 * 
+	 * @return dscAlarmBridgeHandler
+	 */
+	public synchronized DSCAlarmBaseBridgeHandler getDSCAlarmBridgeHandler() {
+		
+		Bridge bridge = getBridge();
+				
+		if (this.dscAlarmBridgeHandler == null) {
+			
+			if (bridge == null) {
+				logger.debug("getDSCAlarmBridgeHandler(): Unable to get bridge!");
+				return null;
+			}
+
+			logger.debug("getDSCAlarmBridgeHandler(): Bridge for '{}' - '{}'",getThing().getUID(), bridge.getUID());
+			
+			ThingHandler handler = bridge.getHandler();
+	
+			if (handler instanceof DSCAlarmBaseBridgeHandler) {
+				this.dscAlarmBridgeHandler = (DSCAlarmBaseBridgeHandler) handler;
+			} else {
+				logger.debug("getDSCAlarmBridgeHandler(): Unable to get bridge handler!");
+			}
+		}
+		
+		return this.dscAlarmBridgeHandler;
+	}
+
+	/**
+	 * Method to Update a Channel
+	 * 
+	 * @param channel
+	 */
+	public abstract void updateChannel(ChannelUID channel);
+
+	/**
+	 * Method to Update Device Properties.
+	 * 
+	 * @param channelUID
+	 * @param state
+	 * @param description
+	 */
+	public abstract void updateProperties(ChannelUID channelUID, int state, String description);
+	
+	/**
+	 * Receives DSC Alarm Events from the bridge.
+	 * 
+	 * @param event.
+	 * @param thing
+	 */
+	public abstract void dscAlarmEventRecieved(EventObject event, Thing thing);
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+	}
+
+	public void onBridgeDisconnected(DSCAlarmBaseBridgeHandler bridge) {
+		logger.debug("onBridgeDisconnected(): Bridge '{}' disconnected",bridge.getThing().getUID());
+		
+		if(bridge.getThing().getUID().equals(getThing().getBridgeUID())) {
+			this.setThingRefreshed(false);
+			updateStatus(ThingStatus.OFFLINE);
```

Comment:
remove this, as it is set by the framework (and then even with the status detail BRIDGE_OFFLINE)